helpers.js

```diff
@@ -0,0 +1,66 @@
+const Rooms = require('./database/Room.js');
+
+let cachedRoom = {};
+const maxReviews = 5;
+
+const findAndCacheRoom = (roomid, callback) => {
+  if (cachedRoom.id === roomid) {
+    callback(null, cachedRoom);
+  } else {
+    Rooms.findOne({ id: roomid }, (err, room) => {
+      if (err) {
+        callback(err);
+      } else {
+        cachedRoom = room;
+        callback(null, cachedRoom);
+      }
+    });
+  }
+};
+
+const findReviewsForGroup = (selReviewGroup, callback) => {
+  let reviewGroup = Number(selReviewGroup);
+  let reviews = [];
+  if (typeof reviewGroup !== 'number') {
+    reviews = cachedRoom.reviews.slice(0, maxReviews);
+  } else {
```

Comment:
This could be cut down even more I think
```suggestion
const findReviewsForGroup = (selReviewGroup = 0, callback) => {
  let reviews = [];
```
 Set the review group to 0 by default and the subsequent logic will take care of the rest.